components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/WSDL20SOAPOperationExtractor.java

```diff
@@ -0,0 +1,193 @@
+
+    }
+
+    public WSDL20SOAPOperationExtractor(WSDLReader wsdlReader) {
+        WSDL20SOAPOperationExtractor.reader = wsdlReader;
+    }
+
+    @Override
+    public boolean init(byte[] wsdlContent) throws APIMgtWSDLException {
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            factory.setNamespaceAware(true);
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            reader.setFeature(WSDLReader.FEATURE_VALIDATION, false);
+            Document dom = builder.parse(new ByteArrayInputStream(wsdlContent));
+            Element domElement = dom.getDocumentElement();
+            WSDLSource wsdlSource = reader.createWSDLSource();
+            wsdlSource.setSource(domElement);
+            wsdlDescription = reader.readWSDL(wsdlSource);
+            canProcess = true;
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully initialized an instance of " + this.getClass().getSimpleName()
+                        + " with a single WSDL.");
+            }
+        } catch (WSDLException | ParserConfigurationException | SAXException | IOException e) {
+            //This implementation class cannot process the WSDL.
+            log.debug("Cannot process the WSDL by " + this.getClass().getName(), e);
+            canProcess = false;
+        }
+        return canProcess;
+    }
+
+    @Override
+    public boolean initPath(String path) throws APIMgtWSDLException {
+        pathToDescriptionMap = new HashMap<>();
+        wsdlArchiveExtractedPath = path;
+        try {
+            WSDLReader reader = getWsdlFactoryInstance().newWSDLReader();
+            reader.setFeature(WSDLReader.FEATURE_VALIDATION, false);
+            File folderToImport = new File(path);
+            Collection<File> foundWSDLFiles = APIFileUtil.searchFilesWithMatchingExtension(folderToImport, "wsdl");
+            if (log.isDebugEnabled()) {
+                log.debug("Found " + foundWSDLFiles.size() + " WSDL file(s) in path " + path);
+            }
+            for (File file : foundWSDLFiles) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Processing WSDL file: " + file.getAbsolutePath());
+                }
+                Description description = reader.readWSDL(file.getAbsolutePath());
+                pathToDescriptionMap.put(file.getAbsolutePath(), description);
+            }
+            if (foundWSDLFiles.size() > 0) {
+                canProcess = true;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully processed all WSDL files in path " + path);
+            }
+        } catch (WSDLException e) {
+            //This implementation class cannot process the WSDL.
+            throw new APIMgtWSDLException("Cannot process the WSDL by " + this.getClass().getName(), e);
```

Comment:
Do we need to log the class name?

Reply:
there are two wsdl processor classes are used for wsdl 1.1 and 2.0. This is added to indicate If one of the wsdl processor classes fails to read the wsdl